arm-iothub/2016-02-03/swagger/iothub.json

```diff
@@ -1040,9 +1040,15 @@
             "$ref": "#/definitions/IpFilterRule"
           }
         },
+        "provisioningState": {
+          "description": "The provisioning state.",
+          "type": "string",
+          "readOnly": true
+        },
         "hostName": {
           "description": "The name of the host.",
-          "type": "string"
+          "type": "string",
+          "readOnly": true
         },
         "eventHubEndpoints": {
```

Comment:
Have all the models been described accurately as far modelling "required" properties of a model is concerned?
One example that I randomly found as incorrect. "IotHubSkuInfo" model does not have any required model properties.
I think atleast name should be required. Based on the current model description the user can provide an empty object `{ }` and that will fail the request.

So please scan all the models and make sure that their properties are accurately marked as "required".

Primary goal: The swagger spec should accurately describe what is happening on the wire.